app/controllers/crops_controller.rb

```diff
@@ -51,18 +51,16 @@ def hierarchy
 
   # GET /crops/search
   def search
-    @search = params[:search]
-    @exact_match = Crop.find_by_name(params[:search])
+    @all_matches = Crop.search(params[:search])
 
-    @partial_matches = Crop.search(params[:search])
-    # exclude exact match from partial match list
-    @partial_matches.reject!{ |r| @exact_match && r.eql?(@exact_match) }
-
-    @fuzzy = Crop.search(params[:term])
+    if exact_match = Crop.find_by_name(params[:search])
```

Comment:
For exact matches, we can probably use [Elasticsearch filters](http://www.elasticsearch.org/guide/en/elasticsearch/reference/current/query-dsl-filters.html), which should perform better than filtering out results in Rails.